demos/interactive_face_detection_demo/detectors.cpp

```diff
@@ -143,63 +143,75 @@ CNNNetwork FaceDetection::read(const InferenceEngine::Core& ie)  {
+                if (op->get_friendly_name() == output) {
+                    auto detOutput = std::dynamic_pointer_cast<ngraph::op::DetectionOutput>(op);
+                    if (!detOutput) {
+                        THROW_IE_EXCEPTION << "Face Detection network output layer(" + op->get_friendly_name() +
+                            ") should be DetectionOutput, but was " +  op->get_type_info().name;
+                    }
+
+                    num_classes = detOutput->get_attrs().num_classes;
+                    break;
                 }
+            }
+        } else {
+            const CNNLayerPtr outputLayer = network.getLayerByName(output.c_str());
+            if (outputLayer->type != "DetectionOutput") {
+                throw std::logic_error("Face Detection network output layer(" + outputLayer->name +
+                                       ") should be DetectionOutput, but was " +  outputLayer->type);
+            }
 
-                num_classes = detOutput->get_attrs().num_classes;
-                break;
+            if (outputLayer->params.find("num_classes") == outputLayer->params.end()) {
+                throw std::logic_error("Face Detection network output layer (" +
+                                       output + ") should have num_classes integer attribute");
             }
+
+            num_classes = outputLayer->GetParamAsUInt("num_classes");
         }
-    } else {
-        const CNNLayerPtr outputLayer = network.getLayerByName(output.c_str());
-        if (outputLayer->type != "DetectionOutput") {
-            throw std::logic_error("Face Detection network output layer(" + outputLayer->name +
-                                   ") should be DetectionOutput, but was " +  outputLayer->type);
+        if (labels.size() != num_classes) {
+            if (labels.size() == (num_classes - 1))  // if network assumes default "background" class, which has no label
+                labels.insert(labels.begin(), "fake");
+            else
+                labels.clear();
         }
-
-        if (outputLayer->params.find("num_classes") == outputLayer->params.end()) {
-            throw std::logic_error("Face Detection network output layer (" +
-                                   output + ") should have num_classes integer attribute");
+        const SizeVector outputDims = _output->getTensorDesc().getDims();
+        maxProposalCount = outputDims[2];
+        objectSize = outputDims[3];
+        if (objectSize != 7) {
+            throw std::logic_error("Face Detection network output layer should have 7 as a last dimension");
+        }
+        if (outputDims.size() != 4) {
+            throw std::logic_error("Face Detection network output dimensions not compatible shoulld be 4, but was " +
+                                   std::to_string(outputDims.size()));
+        }
+        _output->setPrecision(Precision::FP32);
+    } else {
+        for (const auto& outputLayer: outputInfo) {
```

Comment:
You should check that there are only 2 outputs.
if no output found with `outputDims.size() == 2 && outputDims[outputDims.size() - 1] == 5`, an exception should be thrown.
Number of dimensions for `labels` blob should also be verified if the output is used.
I also suggest explicitly setting precision for outputs.

Reply:
Recently added models actually have more than 2 layers, so I would not check exact number of outputs. I'll better add info about unused (they can not be removed from IR) outputs.

The requested exception has been added if required outputs have not been determined

Explicitly set precision for `boxes`